### PR TITLE
Function Context support

### LINKED ISSUE #66: Support writing to Context
### What problem are you facing?

Currently this function allows reading from the Context:
```
{{- $environmentConfig := index .context "apiextensions.crossplane.io/environment" }}
```

But one can not write to the context.

### How could this Function help solve your problem?

Adding a `Context` resource similar to what already done with `CompositeConnectionDetails`, could be an option.
```yaml
apiVersion: meta.gotemplating.fn.crossplane.io/v1alpha1
kind: Context
data:
  apiextensions.crossplane.io/environment: ....
  another.key: ...
```
TBD how to handle it w.r.t. the input Context, we could replace it altogether, but merging could be a little bit hard with this function, probably it's the best thing to do though.

### PR DESCRIPTION
**Pull request by stevendborrelli**
### Description of your changes

Adds support for writing to Context

```
apiVersion: meta.gotemplating.fn.crossplane.io/v1alpha1
kind: Context
  data:
```

Fixes #66 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit tests for my change.

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

## Testing

Given the following template:

```yaml
 ---
          apiVersion: meta.gotemplating.fn.crossplane.io/v1alpha1
          kind: Context
          data:
            "apiextensions.crossplane.io/environment":
               update: environment
               nestedEnvUpdate:
                 hello: world
            "other-context-key":
              complex: {{ index .context "apiextensions.crossplane.io/environment" "complex" | toYaml | nindent 6 }}
            newkey:
              hello: world
          ---
          apiVersion: example.crossplane.io/v1
          kind: XR
          status:
            fromEnv: {{ index .context "apiextensions.crossplane.io/environment" "complex" "c" "d" }} 
```

Render will output

```yaml
---
apiVersion: example.crossplane.io/v1
kind: XR
metadata:
  name: example-xr
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  fromEnv: e
---
apiVersion: render.crossplane.io/v1beta1
fields:
  apiextensions.crossplane.io/environment:
    apiVersion: internal.crossplane.io/v1alpha1
    complex:
      a: b
      c:
        d: e
        f: "1"
    kind: Environment
    nestedEnvUpdate:
      hello: world
    update: environment
  newkey:
    hello: world
  other-context-key:
    complex:
      a: b
      c:
        d: e
        f: "1"
kind: Context
```

### CODE REVIEW

**Comment by stevendborrelli** [comment]
@phisco can you take a look? For the merging I am using mergo right now

**Comment by phisco** [comment]
Awesome! Yes, I was thinking about mergo too, I'd say it looks pretty good, let me take it for a spin tomorrow!

Maybe I'd add some annotation to the context meta resource to allow replacing it instead of merging if needed.

**Comment by phisco** [comment]
opened https://github.com/stevendborrelli/function-go-templating/pull/1 with a few changes

**Comment by stevendborrelli** [comment]
@phisco merged your changes. All looks good.

File: example/context/composition.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      inline:
+        template: |
+          ---
+          apiVersion: meta.gotemplating.fn.crossplane.io/v1alpha1
+          kind: Context
+          data:
+            # update existing EnvironmentConfig by using the "apiextensions.crossplane.io/environment" key
+            "apiextensions.crossplane.io/environment":
+               kind: Environment,
+               apiVersion: internal.crossplane.io/v1alpha1,
```
Comment (phisco) [inline example/context/composition.yaml:38]:
```suggestion
               kind: Environment
               apiVersion: internal.crossplane.io/v1alpha1
```

File: README.md [hunk 2]
```diff
@@ -166,6 +166,42 @@ example:
...
+
+```yaml
+---
+apiVersion: meta.gotemplating.fn.crossplane.io/v1alpha1
+kind: Context
+data:
+  "apiextensions.crossplane.io/environment":
+      update: environment
+      nestedEnvUpdate:
+        hello: world
```
Comment (phisco) [inline README.md:198]:
```suggestion
      kind: Environment
      apiVersion: internal.crossplane.io/v1alpha1
      update: environment
      nestedEnvUpdate:
        hello: world
```

File: example/context/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        d: e
+        f: "1"
+    kind: Environment
+    nestedEnvUpdate:
+      hello: world
+    update: environment
+  newkey:
+    apiVersion: internal.crossplane.io/v1alpha1
+    hello: world
+    kind: Environment
```
Comment (phisco) [inline example/context/README.md:52]:
```suggestion
    hello: world
```

File: example/context/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  other-context-key:
+    apiVersion: internal.crossplane.io/v1alpha1
+    complex:
+      a: b
+      c:
+        d: e
+        f: "1"
+    kind: Environment
```
Comment (phisco) [inline example/context/README.md:60]:
```suggestion
    complex:
      a: b
      c:
        d: e
        f: "1"
```

**Review by phisco — COMMENTED** [review]
sorry forgot to update the examples

**Review by phisco — APPROVED** [review]

**Pull request merged**